media-libs/libid3tag/libid3tag-0.16.3.recipe

```diff
@@ -34,36 +35,27 @@ BUILD_REQUIRES="
 	devel:libz$secondaryArchSuffix
 	"
 BUILD_PREREQUIRES="
-	cmd:aclocal_1.13
-	cmd:autoconf
+	cmd:cmake
 	cmd:gcc$secondaryArchSuffix
 	cmd:gperf >= 3.1
-	cmd:libtoolize$secondaryArchSuffix
-	cmd:make
+	cmd:ninja
 	"
 
 BUILD()
 {
-	touch AUTHORS ChangeLog NEWS
-	autoreconf -vfi
-	runConfigure ./configure --disable-static
-	make
+	cmake -S . -B build -G Ninja \
+		$cmakeDirArgs \
+		-DCMAKE_BUILD_TYPE=Release
+	cmake --build build $jobArgs
 }
 
 INSTALL()
 {
-	make install
-
-	rm $libDir/libid3tag.la
+	cmake --install build
 
 	prepareInstalledDevelLib libid3tag
 	fixPkgconfig
 
 	packageEntries devel \
 		$developDir
 }
-
-TEST()
-{
-	make check
```

Comment:
Are there no tests in the cmake build? Normally, they could be run with ctest.

Reply:
I don't believe there are tests with CMake at the moment, or at least they're not setup currently.

Reply:
You could try with this in TEST()?
````
ctest --test-dir build --output-on-failure
````

Reply:
> You could try with this in TEST()?
> 
> ```
> ctest --test-dir build --output-on-failure
> ```

Here's the full output after adding that to the package recipe:

```
~> haikuporter -S libid3tag-0.16.3
Checking if any dependency-infos need to be updated ...
Looking for stale dependency-infos ...
----------------------------------------------------------------------
media-libs::libid3tag-0.16.3
        /boot/home/haikuports/media-libs/libid3tag/libid3tag-0.16.3.recipe
----------------------------------------------------------------------

Downloading: https://codeberg.org/tenacityteam/libid3tag/archive/0.16.3.tar.gz ...
--2024-11-18 17:57:05--  https://codeberg.org/tenacityteam/libid3tag/archive/0.16.3.tar.gz
Resolving codeberg.org... 2001:67c:1401:20f0::1, 217.197.91.145
Connecting to codeberg.org|2001:67c:1401:20f0::1|:443... failed: Network is unreachable.
Connecting to codeberg.org|217.197.91.145|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 53218 (52K) [application/octet-stream]
Saving to: ‘/boot/home/haikuports/media-libs/libid3tag/download/0.16.3.tar.gz’
     0K                                                      100%  166K=0.3s
2024-11-18 17:57:05 (166 KB/s) - ‘/boot/home/haikuports/media-libs/libid3tag/download/0.16.3.tar.gz’ saved [53218/53218]
Validating checksum of 0.16.3.tar.gz
Unpacking source of 0.16.3.tar.gz
Initialized empty Git repository in /boot/home/haikuports/media-libs/libid3tag/work-0.16.3/sources/libid3tag/.git/
chroot has these packages active:
        /boot/system/packages/bash-5.2.032-1-x86_64.hpkg
        /boot/system/packages/binutils-2.42-1-x86_64.hpkg
        /boot/system/packages/bzip2-1.0.8-3-x86_64.hpkg
        /boot/system/packages/ca_root_certificates-2024_07_02-1-any.hpkg
        /boot/system/packages/cmake-3.30.5-1-x86_64.hpkg
        /boot/system/packages/coreutils-9.3-3-x86_64.hpkg
        /boot/system/packages/curl-8.10.1-3-x86_64.hpkg
        /boot/system/packages/expat-2.6.2-1-x86_64.hpkg
        /boot/system/packages/findutils-4.9.0-2-x86_64.hpkg
        /boot/system/packages/fontconfig-2.13.96-2-x86_64.hpkg
        /boot/system/packages/freetype-2.13.2-1-x86_64.hpkg
        /boot/system/packages/gcc-13.3.0_2023_08_10-2-x86_64.hpkg
        /boot/system/packages/gcc_syslibs-13.3.0_2023_08_10-2-x86_64.hpkg
        /boot/system/packages/gettext_libintl-0.22.5-1-x86_64.hpkg
        /boot/system/packages/gmp-6.3.0-1-x86_64.hpkg
        /boot/system/packages/gperf-3.1-1-x86_64.hpkg
        /boot/system/packages/graphite2-1.3.14-2-x86_64.hpkg
        /boot/system/packages/grep-3.11-2-x86_64.hpkg
        /boot/system/packages/gzip-1.12-1-x86_64.hpkg
        /boot/system/packages/haiku-r1~beta5_hrev57937_125-1-x86_64.hpkg
        /boot/system/packages/haiku_devel-r1~beta5_hrev57937_125-1-x86_64.hpkg
        /boot/system/packages/harfbuzz-8.3.0-2-x86_64.hpkg
        /boot/system/packages/icu74-74.1-5-x86_64.hpkg
        /boot/system/packages/intel_wifi_firmwares-2023_03_06-1-any.hpkg
        /boot/system/packages/jsoncpp-1.9.5-3-x86_64.hpkg
        /boot/system/packages/libarchive-3.7.2-2-x86_64.hpkg
        /boot/system/packages/libedit-20230828_3.1-1-x86_64.hpkg
        /boot/system/packages/libiconv-1.17-4-x86_64.hpkg
        /boot/system/packages/libidn2-2.0.5-3-x86_64.hpkg
        /boot/system/packages/libpcre2-10.43-2-x86_64.hpkg
        /boot/system/packages/libpng16-1.6.44-1-x86_64.hpkg
        /boot/system/packages/libpsl-0.21.5-1-x86_64.hpkg
        /boot/system/packages/libssh2-1.11.1-1-x86_64.hpkg
        /boot/system/packages/libunistring-1.2-1-x86_64.hpkg
        /boot/system/packages/libuv-1.48.0-1-x86_64.hpkg
        /boot/system/packages/libxml2-2.12.9-1-x86_64.hpkg
        /boot/system/packages/mpc-1.2.1-2-x86_64.hpkg
        /boot/system/packages/mpfr-4.2.0-3-x86_64.hpkg
        /boot/system/packages/ncurses6-6.5-2-x86_64.hpkg
        /boot/system/packages/nghttp2-1.63.0-1-x86_64.hpkg
        /boot/system/packages/ninja-1.11.1-2-x86_64.hpkg
        /boot/system/packages/noto-20240201-1-any.hpkg
        /boot/system/packages/noto_sans_cjk_jp-2.004-1-any.hpkg
        /boot/system/packages/openssl3-3.0.14-2-x86_64.hpkg
        /boot/system/packages/ralink_wifi_firmwares-2015_02_11-2-any.hpkg
        /boot/system/packages/readline-8.2.013-1-x86_64.hpkg
        /boot/system/packages/realtek_wifi_firmwares-2019_01_02-1-any.hpkg
        /boot/system/packages/rhash-1.4.4-3-x86_64.hpkg
        /boot/system/packages/sed-4.9-1-x86_64.hpkg
        /boot/system/packages/tar-1.35-2-x86_64.hpkg
        /boot/system/packages/unzip-6.10c23-5-x86_64.hpkg
        /boot/system/packages/xz_utils-5.6.2-2-x86_64.hpkg
        /boot/system/packages/zlib-1.3.1-4-x86_64.hpkg
        /boot/system/packages/zlib_devel-1.3.1-4-x86_64.hpkg
        /boot/system/packages/zstd-1.5.6-1-x86_64.hpkg
----- Package Info ----------------
header size:                     80
heap size:                      716
TOC size:                        41
package attributes size:        634
total size:                     796
-----------------------------------
----- Package Info ----------------
header size:                     80
heap size:                      751
TOC size:                        41
package attributes size:        679
total size:                     831
-----------------------------------
waiting for build package libid3tag-0.16.3-1 to be activated
waiting for build package libid3tag_devel-0.16.3-1 to be activated
Building ...
CMake Deprecation Warning at CMakeLists.txt:1 (cmake_minimum_required):
  Compatibility with CMake < 3.5 will be removed from a future version of
  CMake.

  Update the VERSION argument <min> value or use a ...<max> suffix to tell
  CMake that the project does not need compatibility with older versions.


-- The C compiler identification is GNU 13.3.0
-- The CXX compiler identification is GNU 13.3.0
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /bin/cc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /bin/c++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Looking for sys/stat.h
-- Looking for sys/stat.h - found
-- Looking for unistd.h
-- Looking for unistd.h - found
-- Looking for assert.h
-- Looking for assert.h - found
-- Looking for ftruncate
-- Looking for ftruncate - found
-- Found ZLIB: /boot/system/develop/lib/libz.so (found version "1.3.1")
-- Configuring done (1.5s)
-- Generating done (0.0s)
-- Build files have been written to: /sources/libid3tag/build
[19/19] Creating library symlink libid3tag.so.0 libid3tag.so
Collecting files to be packaged ...
-- Install configuration: "Release"
-- Installing: /packages/libid3tag-0.16.3-1/.self/lib/libid3tag.so.0.16.3
-- Installing: /packages/libid3tag-0.16.3-1/.self/lib/libid3tag.so.0
-- Installing: /packages/libid3tag-0.16.3-1/.self/lib/libid3tag.so
-- Installing: /packages/libid3tag-0.16.3-1/.self/develop/headers/id3tag.h
-- Installing: /packages/libid3tag-0.16.3-1/.self/lib/pkgconfig/id3tag.pc
-- Installing: /packages/libid3tag-0.16.3-1/.self/lib/cmake/id3tag/id3tagTargets.cmake
-- Installing: /packages/libid3tag-0.16.3-1/.self/lib/cmake/id3tag/id3tagTargets-release.cmake
-- Installing: /packages/libid3tag-0.16.3-1/.self/lib/cmake/id3tag/id3tagConfig.cmake
-- Installing: /packages/libid3tag-0.16.3-1/.self/lib/cmake/id3tag/id3tagConfigVersion.cmake
/packages/libid3tag_devel-0.16.3-1/.self
name                    libid3tag
version                 0.16.3-1
architecture            x86_64
summary                 "The MAD id3tag library"
description             "libid3tag is a library for reading and writing ID3 tags, both ID3v1 and the various versions of ID3v2."
packager                "Avery King <avery98@pm.me>"
vendor                  "Haiku Project"
licenses {
        "GNU GPL v2"
}
copyrights {
        "2000-2004 Underbit Technologies, Inc."
        "2021-2024 Tenacity Team"
}
provides {
        libid3tag = 0.16.3
        lib:libid3tag = 0.3.0 compat >= 0
}
requires {
        haiku >= r1~beta5_hrev57937_125-1
        lib:libz >= 1.3.1
}
urls {
        "http://www.underbit.com/products/mad/"
}
source-urls {
# Download
        "https://codeberg.org/tenacityteam/libid3tag/archive/0.16.3.tar.gz"
# Location 2
        "https://ports-mirror.haiku-os.org/libid3tag/0.16.3.tar.gz"
}
mimesetting files for package libid3tag-0.16.3-1-x86_64.hpkg ...
creating package libid3tag-0.16.3-1-x86_64.hpkg ...
----- Package Info ----------------
header size:                     80
heap size:                    38081
TOC size:                       586
package attributes size:        577
total size:                   38161
-----------------------------------
name                    libid3tag_devel
version                 0.16.3-1
architecture            x86_64
summary                 "The MAD id3tag library (development files)"
description             "libid3tag is a library for reading and writing ID3 tags, both ID3v1 and the various versions of ID3v2."
packager                "Avery King <avery98@pm.me>"
vendor                  "Haiku Project"
licenses {
        "GNU GPL v2"
}
copyrights {
        "2000-2004 Underbit Technologies, Inc."
        "2021-2024 Tenacity Team"
}
provides {
        libid3tag_devel = 0.16.3
        devel:libid3tag = 0.3.0 compat >= 0
}
requires {
        libid3tag == 0.16.3 base
}
urls {
        "http://www.underbit.com/products/mad/"
}
source-urls {
# Download
        "https://codeberg.org/tenacityteam/libid3tag/archive/0.16.3.tar.gz"
# Location 2
        "https://ports-mirror.haiku-os.org/libid3tag/0.16.3.tar.gz"
}
mimesetting files for package libid3tag_devel-0.16.3-1-x86_64.hpkg ...
creating package libid3tag_devel-0.16.3-1-x86_64.hpkg ...
----- Package Info ----------------
header size:                     80
heap size:                     3786
TOC size:                       558
package attributes size:        565
total size:                    3866
-----------------------------------
waiting for build package libid3tag-0.16.3-1 to be deactivated
waiting for build package libid3tag_devel-0.16.3-1 to be deactivated
cleaning chroot folder
grabbing libid3tag-0.16.3-1-x86_64.hpkg and moving it to /boot/home/haikuports/packages/libid3tag-0.16.3-1-x86_64.hpkg
grabbing libid3tag_devel-0.16.3-1-x86_64.hpkg and moving it to /boot/home/haikuports/packages/libid3tag_devel-0.16.3-1-x86_64.hpkg
```

Reply:
Running the test should be done with `hp foo --test`. :)

Reply:
Whoops! My bad! I'll try again when I have time 😄

Note that when you try to run CTest when building libid3tag manually, it outputs that there are no tests.

Reply:
No tests provided, this can be skipped.